metadata-ingestion/src/datahub/emitter/sql_parsing_builder.py

```diff
@@ -255,6 +256,8 @@ def _merge_lineage_data(
                 column_map = upstream_edges[upstream_column_info.table].column_map
                 column_map[cl.downstream.column].add(upstream_column_info.column)
```

Comment:
so we're mutating in place, and then explicitly setting the key back to the same object (from above) so that it gets registered by the FileBackedDict?

we should add some comments here calling that out since it's pretty unclear right now :)